datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -332,6 +286,26 @@ def __init__(self, *args, **kwargs):
         if not PY2:
             self.check_initializations.append(self.load_configuration_models)
 
+    def _create_metrics_pattern(self, config, option_name):
+        filters = config.get(option_name, []) if config else []
+        if not isinstance(filters, list):
+            raise ConfigurationError('Setting `{}` must be an array'.format(option_name))
+
+        metrics_pattern = None
+        metrics_patterns = []
+        for i, entry in enumerate(filters, 1):
+            if not isinstance(entry, str):
+                raise ConfigurationError('Entry #{} of setting `{}` must be a string'.format(i, option_name))
+            if not entry:
+                self.log.debug('Entry #%s of setting `%s` must not be empty, ignoring', i, option_name)
+                continue
+
+            metrics_patterns.append(entry)
+
+        if metrics_patterns:
+            metrics_pattern = re.compile('|'.join(metrics_patterns))
+        return metrics_pattern
```

Comment:
Would it make sense to return the values directly, instead of going through a variable?

```suggestion
        if metrics_patterns:
            return re.compile('|'.join(metrics_patterns))
        return None
```

Reply:
Sure, I agree it's a little more readable this way